src/lib.rs

```diff
@@ -1393,28 +1392,34 @@ impl CredentialConfig {
 }
 
 impl Api {
-    pub fn new() -> Result<Api, u32> {
-        let new_table: *const ApiTable = ptr::null();
-        let status = unsafe { MsQuicOpenVersion(2, &new_table) };
-        if Status::failed(status) {
-            return Err(status);
+    pub fn new() -> Self {
+        // We initialize APITABLE at only once.
+        unsafe {
+            START_MSQUIC.call_once(|| {
```

Comment:
I'm confused why this is both in the Api and Registration interface. Aren't they separate 'call once' things?

Reply:
Api may be initialized and its method may be called before registration initilized. But Api may not initialized. So, I suppose that we should call `MsQuicOpenVersion()` when Api OR Registration is initialized.
And both `Api::new()` and `Registration::new()` will access the same `START_MSQUIC`. If Api is initialized before Registration is initialized, the closure which is passed to `START_MSQUIC.call_once` in `Registration::new` will not be executed because `START_MSQUIC.call_once` is already executed in `Api::new()`. If Registration is first initialized, its closure will be executed.

Reply:
@nibanks Let me know what you think about this.